scripts/run.sh

```diff
@@ -137,38 +137,42 @@ RECO_DIR=RECO/${TAG}
 RECO_TEMP=${TMPDIR}/${RECO_DIR}
 mkdir -p ${RECO_TEMP} 
 
-# Mix background events
-{
-  date
-  eic-info
-  prmon \
-    --filename ${LOG_TEMP}/${TASKNAME}.hepmcmerger.prmon.txt \
-    --json-summary ${LOG_TEMP}/${TASKNAME}.hepmcmerger.prmon.json \
-    -- \
-  SignalBackgroundMerger \
-    --rngSeed ${SEED:-1} \
-    --nSlices ${EVENTS_PER_TASK} \
-    --signalSkip ${SKIP_N_EVENTS} \
-    --signalFile ${INPUT_FILE} \
-    --bg1Freq ${BG1_FREQ:-""} \
-    --bg1File ${BG1_FILE:-""} \
-    --bg1Skip ${BG1_SKIP:-0} \
-    --bg2Freq ${BG2_FREQ:-""} \
-    --bg2File ${BG2_FILE:-""} \
-    --bg2Skip ${BG2_SKIP:-0} \
-    --bg3Freq ${BG3_FREQ:-""} \
-    --bg3File ${BG3_FILE:-""} \
-    --bg3Skip ${BG3_SKIP:-0} \
-    --bg4Freq ${BG4_FREQ:-""} \
-    --bg4File ${BG4_FILE:-""} \
-    --bg4Skip ${BG4_SKIP:-0} \
-    --outputFile ${FULL_TEMP}/${TASKNAME}.hepmc3.tree.root
+# Mix background events if the input file is a hepmc file
+if [[ "$EXTENSION" == "hepmc3.tree.root" ]]; then
+  {
+    date
+    eic-info
+    prmon \
+      --filename ${LOG_TEMP}/${TASKNAME}.hepmcmerger.prmon.txt \
+      --json-summary ${LOG_TEMP}/${TASKNAME}.hepmcmerger.prmon.json \
+      -- \
+    SignalBackgroundMerger \
+      --rngSeed ${SEED:-1} \
+      --nSlices ${EVENTS_PER_TASK} \
+      --signalSkip ${SKIP_N_EVENTS} \
+      --signalFile ${INPUT_FILE} \
+      --bg1Freq ${BG1_FREQ:-""} \
+      --bg1File ${BG1_FILE:-""} \
+      --bg1Skip ${BG1_SKIP:-0} \
+      --bg2Freq ${BG2_FREQ:-""} \
+      --bg2File ${BG2_FILE:-""} \
+      --bg2Skip ${BG2_SKIP:-0} \
+      --bg3Freq ${BG3_FREQ:-""} \
+      --bg3File ${BG3_FILE:-""} \
+      --bg3Skip ${BG3_SKIP:-0} \
+      --bg4Freq ${BG4_FREQ:-""} \
+      --bg4File ${BG4_FILE:-""} \
+      --bg4Skip ${BG4_SKIP:-0} \
+      --outputFile ${FULL_TEMP}/${TASKNAME}.hepmc3.tree.root
 
-  # Use background merged file as input for next stage
-  INPUT_FILE=${FULL_TEMP}/${TASKNAME}.hepmc3.tree.root
-  # Don't skip events on the background merged file
-  SKIP_N_EVENTS=0
-} 2>&1 | tee ${LOG_TEMP}/${TASKNAME}.hepmcmerger.log | tail -n1000
+    # Use background merged file as input for next stage
+    INPUT_FILE=${FULL_TEMP}/${TASKNAME}.hepmc3.tree.root
+    # Don't skip events on the background merged file
+    SKIP_N_EVENTS=0
+  } 2>&1 | tee ${LOG_TEMP}/${TASKNAME}.hepmcmerger.log | tail -n1000
+else
+  echo "No background mixing is performed for singles"
```

Comment:
Tee into output log file to ensure it exists (or make sure upload can handle missing files in case of singles).

Reply:
We are not uploading the hepmc merger log

Reply:
Shouldn't we? I don't know what it includes but it might be useful if we need to start figuring out why things don't look the way they should.

Reply:
I guess it doesn't hurt to have it. I was thinking we wouldn't get much extra information from it and inconsistencies in merging would be apparent from looking at the primaries in the output file.  I will add it in for the background mixed case.